src/plugins/discover/public/application/embeddable/saved_search_embeddable.tsx

```diff
@@ -344,64 +189,227 @@ export class SearchEmbeddable
+    searchProps: SearchProps,
     { forceFetch = false }: { forceFetch: boolean } = { forceFetch: false }
   ) {
     const isFetchRequired =
       !esFilters.onlyDisabledFiltersChanged(this.input.filters, this.prevFilters) ||
-      !_.isEqual(this.prevQuery, this.input.query) ||
-      !_.isEqual(this.prevTimeRange, this.input.timeRange) ||
-      !_.isEqual(searchScope.sort, this.input.sort || this.savedSearch.sort) ||
+      !isEqual(this.prevQuery, this.input.query) ||
+      !isEqual(this.prevTimeRange, this.input.timeRange) ||
+      !isEqual(searchProps.sort, this.input.sort || this.savedSearch.sort) ||
       this.prevSearchSessionId !== this.input.searchSessionId;
 
     // If there is column or sort data on the panel, that means the original columns or sort settings have
     // been overridden in a dashboard.
-    searchScope.columns = handleSourceColumnState(
+    searchProps.columns = handleSourceColumnState(
       { columns: this.input.columns || this.savedSearch.columns },
       this.services.core.uiSettings
     ).columns;
+
     const savedSearchSort =
       this.savedSearch.sort && this.savedSearch.sort.length
         ? this.savedSearch.sort
         : getDefaultSort(
-            this.searchScope?.indexPattern,
+            this.searchProps?.indexPattern,
             getServices().uiSettings.get(SORT_DEFAULT_ORDER_SETTING, 'desc')
           );
-    searchScope.sort = this.input.sort || savedSearchSort;
-    searchScope.sharedItemTitle = this.panelTitle;
-
+    searchProps.sort = this.input.sort || savedSearchSort;
+    searchProps.sharedItemTitle = this.panelTitle;
     if (forceFetch || isFetchRequired) {
-      this.filtersSearchSource!.setField('filter', this.input.filters);
-      this.filtersSearchSource!.setField('query', this.input.query);
+      this.filtersSearchSource.setField('filter', this.input.filters);
+      this.filtersSearchSource.setField('query', this.input.query);
       if (this.input.query?.query || this.input.filters?.length) {
-        this.filtersSearchSource!.setField('highlightAll', true);
+        this.filtersSearchSource.setField('highlightAll', true);
       } else {
-        this.filtersSearchSource!.removeField('highlightAll');
+        this.filtersSearchSource.removeField('highlightAll');
       }
 
       this.prevFilters = this.input.filters;
       this.prevQuery = this.input.query;
       this.prevTimeRange = this.input.timeRange;
       this.prevSearchSessionId = this.input.searchSessionId;
-      this.fetch();
-    } else if (this.searchScope) {
-      // trigger a digest cycle to make sure non-fetch relevant changes are propagated
-      this.searchScope.$applyAsync();
+      this.searchProps = searchProps;
+      await this.fetch();
+      if (this.node) {
+        await this.rerenderComponent(this.node);
+      }
+    } else if (this.searchProps && this.node) {
+      this.searchProps = searchProps;
+      this.renderReactComponent(this.node, this.searchProps);
+    }
```

Comment:
Since rendering is triggered in any case in this code, and `rerenderComponent` can trigger `rerenderComponent` by `pushContainerStateParamsToProps` I'd suggest to simplify my moving to just use of `renderReactComponent` like this:
```suggestion
    } else if (this.searchProps && this.node) {
      this.searchProps = searchProps;
    }
    if (this.node) {
      this.renderReactComponent(this.node, this.searchProps!);
    }  
```
 ... and `rerenderComponent` could be deleted